operator/cmd/mesh/install.go

```diff
@@ -136,12 +138,17 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, iArgs *installArgs, log
 	if err != nil {
 		return err
 	}
+	setFlags := applyFlagAliases(iArgs.set, iArgs.manifestsPath, iArgs.revision)
 	// Ignore the err because we don't want to show
 	// "no running Istio pods in istio-system" for the first time
 	_ = DetectIstioVersionDiff(cmd, tag, kubeClient, iArgs)
 	// Warn users if they use `istioctl install` without any config args.
 	if !rootArgs.dryRun && !iArgs.skipConfirmation {
-		prompt := fmt.Sprintf("This will install the Istio %s profile into the cluster. Proceed? (y/N)", tag)
+		profile, enabledComponents, err := getProfileAndEnabledComponents(setFlags, iArgs.inFilenames, iArgs.force, iArgs.kubeConfigPath, iArgs.context, l)
+		if err != nil {
+			return fmt.Errorf("failed to get profile and enabled components: %v", err)
+		}
+		prompt := fmt.Sprintf("This will install the Istio %s profile %s with %q into the cluster. Proceed? (y/N)", tag, profile, enabledComponents)
```

Comment:
can we have profile name before profile?  e.g. This will install the Istio 1.9.0 default profile...

Reply:
with ["Base" "Pilot" "Cni" "IngressGateways"] components

Reply:
I think this is a great improvement.

> with ["Base" "Pilot" "Cni" "IngressGateways"] components

But not sure if we are making the install message more complicated. eg: if I use `profile=demo`, After the installation is complete we only see. `istiod`, `ingress`, and `egress` components and not the `Base` and `Pilot` components, which might be confusing to the users?

```
$ kubectl get pods -n istio-system
NAME                                    READY   STATUS              RESTARTS   AGE
istio-egressgateway-77989d66b7-rdqlk    0/1     Running             0          28m
istio-ingressgateway-75bf97b98d-rw8g5   0/1     Running             0          28m
istiod-1-7-3-6cf75cfbd5-5jg27           0/1     Running             0          33m
```

Reply:
@shamsher31 We are simply retrieve components from operator API to make sure the end users knows what components they are going to install, IMO, the mapping between component and deployments should be consider in future commit.